compositor_render/src/transformations/builtin/box_layout.rs

```diff
@@ -10,6 +14,45 @@ pub struct BoxLayout {
 }
 
 impl BoxLayout {
+    /// Returns box representing position of input frame fitted into `self`,     
+    /// Fitted means scaled to input resolution without cropping or changing aspect ratio
+    pub fn fit(
```

Comment:
This is great, but I worry a bit about how this will scale, you will add fit and stretch next etc. maybe some other transform too.

If fit and stretch are the only parts we will ever add here then it's fine, but if we are planning to add anything else I would suggest a different abstraction e.g. trait BoxLayoutTransformation or maybe just a standalone function that are not associated to the BoxLayout type

Reply:
Yep, so far I haven't planned to add anything other than `fit`/`fill` here (`stretch` happens automatically). I think we can break it down once we need to add more stuff here (probably into functions).